pkg/api/org_test.go

```diff
@@ -37,13 +40,170 @@ var (
 	testCreateOrgCmd = `{ "name": "TestOrg%v"}`
 )
 
-func TestAPIEndpoint_CreateOrgs_LegacyAccessControl(t *testing.T) {
-	sc := setupHTTPServer(t, false)
+// `/api/org` endpoints test
+
+func TestAPIEndpoint_GetCurrentOrg_LegacyAccessControl(t *testing.T) {
+	sc := setupHTTPServer(t, true, false)
+	setInitCtxSignedInViewer(sc.initCtx)
+
+	_, err := sc.db.CreateOrgWithMember("TestOrg", testUserID)
+	require.NoError(t, err)
+
+	t.Run("Viewer can view CurrentOrg", func(t *testing.T) {
+		response := callAPI(sc.server, http.MethodGet, getCurrentOrgURL, nil, t)
+		assert.Equal(t, http.StatusOK, response.Code)
+	})
+
+	sc.initCtx.IsSignedIn = false
+	t.Run("Unsigned user cannot view CurrentOrg", func(t *testing.T) {
+		response := callAPI(sc.server, http.MethodGet, getCurrentOrgURL, nil, t)
+		assert.Equal(t, http.StatusUnauthorized, response.Code)
+	})
+}
+
+func TestAPIEndpoint_GetCurrentOrg_AccessControl(t *testing.T) {
+	sc := setupHTTPServer(t, true, true)
+	setInitCtxSignedInViewer(sc.initCtx)
+
+	_, err := sc.db.CreateOrgWithMember("TestOrg", testUserID)
+	require.NoError(t, err)
+
+	t.Run("AccessControl allows viewing CurrentOrg with correct permissions", func(t *testing.T) {
+		setAccessControlPermissions(sc.acmock, []*accesscontrol.Permission{{Action: ActionOrgsRead}}, sc.initCtx.OrgId)
+		response := callAPI(sc.server, http.MethodGet, getCurrentOrgURL, nil, t)
+		assert.Equal(t, http.StatusOK, response.Code)
+	})
+	t.Run("AccessControl prevents viewing CurrentOrg with correct permissions in another org", func(t *testing.T) {
+		setAccessControlPermissions(sc.acmock, []*accesscontrol.Permission{{Action: ActionOrgsRead}}, 2)
+		response := callAPI(sc.server, http.MethodGet, getCurrentOrgURL, nil, t)
+		assert.Equal(t, http.StatusForbidden, response.Code)
+	})
+	t.Run("AccessControl prevents viewing CurrentOrg with incorrect permissions", func(t *testing.T) {
+		setAccessControlPermissions(sc.acmock, []*accesscontrol.Permission{{Action: "orgs:invalid"}}, sc.initCtx.OrgId)
+		response := callAPI(sc.server, http.MethodGet, getCurrentOrgURL, nil, t)
+		assert.Equal(t, http.StatusForbidden, response.Code)
+	})
+}
+
+func TestAPIEndpoint_PutCurrentOrg_LegacyAccessControl(t *testing.T) {
+	sc := setupHTTPServer(t, true, false)
+
+	_, err := sc.db.CreateOrgWithMember("TestOrg", testUserID)
+	require.NoError(t, err)
+
+	input := strings.NewReader(testUpdateOrgNameForm)
+
+	setInitCtxSignedInViewer(sc.initCtx)
+	t.Run("Viewer cannot update current org", func(t *testing.T) {
```

Comment:
Yes, technically it would also work with an `OrgAdmin` since legacy access-control only check for the `isGrafanaAdmin` boolean. I think that's alright though, we just check that a user with the `isGrafanaAdmin` boolean (even a `Viewer`) can and a user without cannot.

Reply:
I think these are the current org endpoints, so there we check for org admin, no?

And I think I wasn't clear, sorry. My thinking is that it'd make sense to select the highest level role that shouldn't have access to the endpoint to test that the access is not granted to it. For instance, here it could be the case that org editors can actually call this endpoint (which we do not want!) as we're only checking that viewers don't have access to it. But that's a very minor point, just thought I'd share my thoughts :)